controllers/configurationpolicy_utils.go

```diff
@@ -346,7 +346,16 @@ func isAutogenerated(key string) (result bool) {
 func formatTemplate(unstruct unstructured.Unstructured, key string) (obj interface{}) {
 	if key == "metadata" {
 		//nolint:forcetypeassert
-		metadata := unstruct.Object[key].(map[string]interface{})
+		var metadata map[string]interface{}
+		if unstruct.Object[key] != nil {
+			metadata = unstruct.Object[key].(map[string]interface{})
+		} else {
+			log.Info("metadata was nil, add minimum to avoid panic")
```

Comment:
Could you please explain the scenario where this introduces a panic?

Reply:
In case, without `name` key, metadata is nil, 
`complianceType: musthave
      objectDefinition:
        apiVersion: v1
        kind: Pod
        metadata:`
metadata convert to interface{}, so miss matching type, causes err